### PR TITLE
[dune] Provide workspace file with all supported Ocaml builds.

### PR DESCRIPTION
**Pull request by ejgallego**

### CODE REVIEW

File: dev/dune-workspace.all [hunk 1]
```diff
@@ -0,0 +1,14 @@
+(lang dune 1.2)
+
+; Add custom flags here. Default developer profile is `dev`
+(env
+ (dev     (flags :standard -rectypes -w -9-27-50))
+ (release (flags :standard -rectypes)))
+
+(context (opam (switch 4.02.3)))
+(context (opam (switch 4.02.3+32bit)))
```
Comment (SkySkimmer) [inline dev/dune-workspace.all]: Not any more I believe.

File: dev/dune-workspace.all [hunk 2]
```diff
@@ -0,0 +1,14 @@
+(lang dune 1.2)
+
+; Add custom flags here. Default developer profile is `dev`
```
Comment (Zimmi48) [inline dev/dune-workspace.all]: Maybe remove this comment that was copy-pasted.

File: dev/dune-workspace.all [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+(env
+ (dev     (flags :standard -rectypes -w -9-27-50))
+ (release (flags :standard -rectypes)))
+
+(context (opam (switch 4.02.3)))
+(context (opam (switch 4.02.3+32bit)))
+; (context (opam (switch 4.05.0)))
+; (context (opam (switch 4.05.0+32bit)))
```
Comment (Zimmi48) [inline dev/dune-workspace.all]: Why is this commented? And isn't 4.06 missing?

File: dev/dune-workspace.all (hunk 1 above)
Comment (ejgallego) [reply dev/dune-workspace.all]: Fixed.

File: dev/dune-workspace.all (hunk 3 above)
Comment (ejgallego) [reply dev/dune-workspace.all]: It was commented to reflect the already done move. I've amended the commit message to indicate this file compiles Coq for all versions that are tested in the CI, not for all the supported ones, which are IMO too many.

Of course one can just add more versions there.

**Comment by Zimmi48** [comment]
You have a typo in your commit message that you may want to fix (same one that I fixed in the PR title).

**Comment by Zimmi48** [comment]
Not testing because OPAM but yeah, let's merge.

File: dev/dune-workspace.all (hunk 2 above)
Comment (ejgallego) [reply dev/dune-workspace.all]: You are correct in that we should instead define the flags in common file and use `(include ...)`, added to the todo list.

**Comment by Zimmi48** [comment]
@ejgallego Did you push after I commented? It seems that I merged an earlier version of this PR and we are going to have to close it manually. Or try force-pushing the version that I merged 65da59b. It might then close normally.

**Comment by ejgallego** [comment]
> @ejgallego Did you push after I commented? It seems that I merged an earlier version of this PR and we are going to have to close it manually. Or try force-pushing the version that I merged [65da59b](https://github.com/coq/coq/commit/65da59b0bd464d67e092a61bf281fcf62fa987cb). It might then close normally.

Ups indeed sorry for the race condition :S I will force push the old version.

**Comment by ejgallego** [comment]
> Did you force-push before or after closing? Because my advice was to force-push before closing. If you did this, well, that didn't work unfortunately.

I did force-push, the PR closed itself.

**Pull request closed without merging**